src/v/raft/vote_stm.h

```diff
@@ -84,6 +88,10 @@ class vote_stm {
             }
 
             if (value->has_value()) {
+                if (value->value().term != _vote_stm._req.term) {
```

Comment:
I wonder if `!=` should be `>`? Or is `<` impossible because the other side will always update its term?

Reply:
exactly, it is impossible in geral we should never treat replies coming from different ter as succesfull